### PR TITLE
Fixes #20418 - Add taxonomies to newly imported puppet env

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

File: app/services/puppet_class_importer.rb
Comment (dLobatog) [inline app/services/puppet_class_importer.rb:182]: Why is this necessary? With this, no error is shown to the user - just "No new environments created". Before, this would attempt to create a duplicate environment - but it would show the error "Name already exists (even if not visible to current user)"

**Review by dLobatog — CHANGES_REQUESTED** [review]
@xprazak2 Generally 👍 , I was a bit confused at the change to `to_be_created_environments`, is it necessary?

Thanks!

File: app/services/puppet_class_importer.rb
Comment (xprazak2) [reply app/services/puppet_class_importer.rb:182]: I added the `unscoped` exactly because of the error. When you have no environments and you get the error, import just seems broken to the user. You are also leaking information out of taxonomy scope - now you know there is an environment that you are not supposed to see and you know its name.

Import should not try to create duplicates IMHO, because resources are shared. So if there is already an env in db but we do not have it in our taxonomy, then we are not supposed to see it or touch it.

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
[test katello] - will merge if it passes

**Comment by dLobatog** [comment]
Thanks @xprazak2 !

**Comment by ares** [comment]
Thanks @dLobatog for merging, please don't forget to set redmine release field after merge. I've set it to 1.16.0.

**Pull request merged**
